server/src/main/java/org/apache/iotdb/db/engine/tsfilemanagement/level/LevelTsFileManagement.java

```diff
@@ -0,0 +1,508 @@
+            writer.getIOWriterOut().truncate(offset - 1);
+            writer.close();
+            HotCompactionUtils
+                .merge(new TsFileResource(targetFile), getTsFileList(isSeq), storageGroupName,
+                    new HotCompactionLogger(storageGroupDir, storageGroupName), deviceSet, isSeq);
+            if (isSeq) {
+              for (List<TsFileResource> currMergeFile : sequenceTsFileResources) {
+                deleteLevelFiles(currMergeFile);
+              }
+            } else {
+              for (List<TsFileResource> currMergeFile : unSequenceTsFileResources) {
+                deleteLevelFiles(currMergeFile);
+              }
+            }
+          }
+        } else {
+          TsFileResource targetResource = new TsFileResource(targetFile);
+          RestorableTsFileIOWriter writer = new RestorableTsFileIOWriter(targetFile);
+          if (sourceFileList.isEmpty()) {
+            return;
+          }
+          int level = getMergeLevel(sourceFileList.get(0));
+          if (!isMergeFinished) {
+            if (deviceSet.isEmpty()) {
+              Files.delete(targetFile.toPath());
+            } else {
+              writer.getIOWriterOut().truncate(offset - 1);
+              writer.close();
+              if (isSeq) {
+                HotCompactionUtils
+                    .merge(targetResource, sequenceTsFileResources.get(level),
+                        storageGroupName,
+                        new HotCompactionLogger(storageGroupDir, storageGroupName), deviceSet,
+                        true);
+                deleteLevelFiles(sequenceTsFileResources.get(level));
+                sequenceTsFileResources.get(level + 1).add(targetResource);
+              } else {
+                HotCompactionUtils
+                    .merge(targetResource, unSequenceTsFileResources.get(level),
+                        storageGroupName,
+                        new HotCompactionLogger(storageGroupDir, storageGroupName), deviceSet,
+                        false);
+                deleteLevelFiles(unSequenceTsFileResources.get(level));
+                unSequenceTsFileResources.get(level + 1).add(targetResource);
+              }
+            }
+          }
+        }
+      }
+    } catch (IOException e) {
+      logger.error("recover vm error ", e);
+    } finally {
+      if (logFile.exists()) {
+        try {
+          Files.delete(logFile.toPath());
+        } catch (IOException e) {
+          logger.error("delete vm log file error ", e);
+        }
+      }
+    }
+  }
+
+  @Override
+  public void forkCurrentFileList() {
+    forkTsFileList(forkedSequenceTsFileResources, sequenceTsFileResources);
+    forkTsFileList(forkedUnSequenceTsFileResources, unSequenceTsFileResources);
+  }
+
+  private void forkTsFileList(List<List<TsFileResource>> forkedSequenceTsFileResources,
+      List<List<TsFileResource>> sequenceTsFileResources) {
```

Comment:
change the parameter name. it can be either sequenceTsFileResources or unSequenceTsFileResources